src/components/ui/Chat/ChannelList/ChannelItem.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { HTMLAttributes } from 'react';
+
+import { StyledChannelItem } from './Styled';
+import { useChannelListContext } from './ChannelListContext';
+import PopOver from '../../PopOver';
+import PopOverItem, { PopOverItemProps } from '../../PopOver/PopOverItem';
+import Button from '../../Button';
+import { BaseProps } from '../../Base';
+import { Dots } from '../../icons';
+
+export interface ChannelItemProps extends Omit<HTMLAttributes<HTMLLIElement & HTMLLIElement>, 'css'>, BaseProps {
+  name: string;
+  arn: string;
+  moreOptions?: PopOverItemProps[];
+}
+
+export const ChannelItem = (props: ChannelItemProps) => {
+  const { name, arn, moreOptions } = props;
+  const context = useChannelListContext();
```

Comment:
Not sure how I feel about tying a UI component to a context provider. Maybe this is something the consumer should manage

Reply:
What's the draw back? The reason to use context here is so the `ChannelList` container can render children implicitly and manage which `ChannelItem` is active. I thought this would be a lightweight solution to accomplish this.

Reply:
The two drawbacks I see is that the UI is now tied to some piece of context/state. So if you wanted to use this UI in another scenario, you couldn't

The other is a matter of responsibility. It seems like it's getting a bit close to managing application state

Just something to be aware of, and depends on how you see the components/SDK integration working with external developers' apps. I'm good with approving if you're already considering this and think this is a good solution

Reply:
I understand what you mean. I guess this state management is pretty much the main reason this `channelList` component exists. Otherwise, it's just a list of buttons with slightly different styling. Also, nothing would break if a developer just didn't use the `activeChannelArn` property. The other solution I see to this is passing an array of children to the channelList and managing the `activeChannel` directly inside the `channelList` component. I know we also prefer implicitly rendered children. 

If you think there is a better solution, I'm up for it, or if you think we should generate the child elements explicitly, we could do that. Let me know.